src/app/CommandHandler.h

```diff
@@ -559,13 +557,17 @@ class CommandHandler : public Messaging::ExchangeDelegate
     CommandPathRegistry * mCommandPathRegistry = &mBasicCommandPathRegistry;
     Optional<uint16_t> mRefForResponse;
 
+    // A list of InvokeResponseMessages to be sent out by CommandHandler.
```

Comment:
This is a single handle, not a list. Not sure we should have a plural there, or claim it's a list.

Reply:
But it is a linked list. Just the next PR will add more than one element

Reply:
```
void PacketBuffer::AddToEnd(PacketBufferHandle && aPacketHandle)
{
    // Ownership of aPacketHandle's buffer is transferred to the end of the chain.
    PacketBuffer * aPacket = std::move(aPacketHandle).UnsafeRelease();

#if CHIP_SYSTEM_CONFIG_USE_LWIP
    pbuf_cat(this, aPacket);
#else  // !CHIP_SYSTEM_CONFIG_USE_LWIP
    PacketBuffer * lCursor = this;

    while (true)
    {
        uint16_t old_total_length = lCursor->tot_len;
        lCursor->tot_len          = static_cast<uint16_t>(lCursor->tot_len + aPacket->tot_len);
        VerifyOrDieWithMsg(lCursor->tot_len >= old_total_length, chipSystemLayer, "buffer chain too large");
        if (!lCursor->HasChainedBuffer())
        {
            lCursor->next = aPacket;
            break;
        }

        lCursor = lCursor->ChainedBuffer();
    }
#endif // !CHIP_SYSTEM_CONFIG_USE_LWIP
}
```

The total length of the buffer is updated. It's internally a linked-list to allow zero-copy, but there are some edges of the API that do become interesting. I don't think you can "just get the next block stored" with the API. You can just "get N bytes from the front". So you don't get to tell how big the chunks are, and see them as chunks. It's just that chunks added are not copied, and doled-out from the chain when you `ConsumeHead(size_t length)`.

Reply:
Just confirming that after our offline discussion that you are okay with how I am simply copying the WriteClient, and not calling `ConsumeHead` so I should be okay, and don't need to implement my own linked list

Reply:
Yes, for now we are OK.